lib/matplotlib/tests/test_axes.py

```diff
@@ -2781,6 +2783,48 @@ def test_violinplot_bad_widths():
         ax.violinplot(data, positions=range(4), widths=[1, 2, 3])
 
 
+def test_violinplot_bad_quantiles():
+    ax = plt.axes()
+    # First 9 digits of frac(sqrt(73))
+    np.random.seed(544003745)
+    data = [np.random.normal(size=100)]
+
+    # Different size quantile list and plots
+    with pytest.raises(ValueError):
+        ax.violinplot(data, quantiles=[[0.1, 0.2], [0.5]])
```

Comment:
Actually, this API doesn't quite make sense. You allow `quantiles=[[0.1, 0.9], [0.2, 0.8]]` but not `quantiles=[[0.1, 0.2], [0.5]]`.

Either there is just a simple case that has the same quantiles for every postion. Or one can freely define the quantiles per position. Just allowing the same number of quantiles but with different values is an odd cut.

That said, I'm unclear if there is a practical need for different quantiles per position. It prevents comparability of the quantile lines between positions.

Reply:
I don't think it is odd. I feel it perfectly makes sense to allow user draw different quantiles for different violins. I think in this implementation user still can draw same quantiles for each violin easily. But if only support same quantiles for every postion, it would be hard if users really want to draw different quantiles for different violins.

Reply:
But if the quantiles can be different, why should you prohibit different numbers of quantiles between violins?

Reply:
No, this example is user want to plot only one violin, but given two list of percentiles.

Reply:
If you look at the updated code in example folder, my design accepts two violins have different number and value of quantiles. (ie. [0.1, 0.2] for first violin, [0.5, 0.7, 0.9] for second violin)

Reply:
Sorry, misinterpreted the test. Can you please use the same number of quantiles for both lists. Makes it hopefully more obvious that the number of elements in the sublists is not the issue to test.

Reply:
Sure, I will update it with some other changes you mentioned/suggested.